src/bash/patches/0001-Add-plugin-support-to-bash.patch

```
@@ -0,0 +1,804 @@
++    if ( (plugin_handle = dlopen(plugin_path, RTLD_LAZY)) == NULL) {
++#ifdef DEBUG
++        itrace("Plugin: can't load plugin %s: %s\n", plugin_path, dlerror());
++#endif
++        return;
++    }
++
++    /* Check if plugin support shell execve method */
++    on_shell_execve_t* plugin_on_shell_execve_handle = dlsym(plugin_handle, on_shell_execve_function_name);
++    if (dlerror() != NULL) {
++        dlclose(plugin_handle);
++
++#ifdef DEBUG
++        itrace("Plugin: can't find on_shell_execve function %s: %s\n", plugin_path, dlerror());
++#endif
++        return;
++    }
++    
++
++    /* Check if plugin support un-initialization method */
++    plugin_uninit_t* plugin_uninit_handle = dlsym(plugin_handle, plugin_uninit_function_name);
++    if (dlerror() != NULL) {
++        dlclose(plugin_handle);
++
++#ifdef DEBUG
++        itrace("Plugin: can't find plugin_uninit function %s: %s\n", plugin_path, dlerror());
++#endif
++        return;
++    }
++
++    /* Check if plugin support initialization method */
++    plugin_init_t* plugin_init_handle = dlsym(plugin_handle, plugin_init_function_name);
++    if (dlerror() != NULL) {
++        dlclose(plugin_handle);
++
++#ifdef DEBUG
++        itrace("Plugin: can't find plugin_init function %s: %s\n", plugin_path, dlerror());
++#endif
++        return;
++    }
++    else {
++        /* Initialize plugin */
++        plugin_init_handle();
++    }
++
++    /* Add plugin to plugin list */
++    int plugin_load_result = append_plugin(plugin_handle,
++                    plugin_on_shell_execve_handle,
++                    plugin_init_handle,
++                    plugin_uninit_handle);
++
++    if (plugin_load_result == PLUGIN_LOAD_SUCCESS) {
++#ifdef DEBUG
++        itrace("Plugin: plugin %s loaded\n", plugin_path);
++#endif
++    }
++    else {
++        /* Output plugin load error message. */
++        fprintf(stderr, "Plugin: plugin %s load failed, result: %d\n", plugin_path, plugin_load_result);
```

>fprintf [](http://example.com/codeflow?start=9&length=7)

We need to add syslog. Since the error is silently ignored, we need syslog to monitor.

---

Fixed.